### PR TITLE
Only use deno with the extensions feature

### PR DESCRIPTION
**Pull request by kylewillmon**
Unfortunately, it looks like the deno crates we are using fail to build
on x86_64-unknown-linux-musl... That is now recorded in #426, but in the
meantime, this change avoids including deno in our default builds

While making this change, I decided to use the `dep:` syntax for features which was introduced in Rust 1.60. I have updated our MSRV and docs accordingly.

### CODE REVIEW

**Comment by kylewillmon** [comment]
This should fix the Release workflow. To check this, [here is a manually kicked off run](https://github.com/phylum-dev/cli/actions/runs/2387384796)

File: cli/Cargo.toml [hunk 1]
```diff
@@ -1,14 +1,15 @@
 [package]
 name = "phylum-cli"
 version = "3.5.0"
-authors = ["Eric Freitag <eric@phylum.io>"]
-edition = "2018"
+authors = ["Phylum, Inc. <engineering@phylum.io>"]
```
Comment (maxrake) [inline cli/Cargo.toml:4]: 👍 
Nothing against Eric, but this is the correct move...especially for a public repo.

File: cli/Cargo.toml [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
+edition = "2021"
+rust-version = "1.60"
```
Comment (maxrake) [inline cli/Cargo.toml:6]: Does this change warrant a reminder to devs to rustup to this version?

File: docs/building.md [hunk 3]
```diff
@@ -4,7 +4,7 @@ category: 6255e67693d5200013b1fa41
 hidden: false
 ---
 
-Phylum is written in Rust, so you'll need a recent Rust installation to build it (we tested with v1.58.0). [Install Rust](https://www.rust-lang.org/tools/install)
+Phylum is written in Rust, so you'll need a recent Rust installation to build it (we tested with v1.61.0). [Install Rust](https://www.rust-lang.org/tools/install)
```
Comment (maxrake) [inline docs/building.md:7]: v1.61.0 is slightly newer than the v1.60 specified in `cli/cargo.toml`. The guess is that this is intentional (testing was done on a newer version than the minimum one *required*), but pointing it out anyway.

**Review by maxrake — APPROVED** [review]
Sorry for the delay...broken builds are serious...was out for a bit...
Some questions are posed but nothing important enough to hold things up.
LGTM.

File: docs/building.md (hunk 3 above)
Comment (kylewillmon) [reply docs/building.md:7]: Yep. 1.61 is what I've got installed, so it's what I tested with.

File: cli/Cargo.toml (hunk 2 above)
Comment (kylewillmon) [reply cli/Cargo.toml:6]: The hope of adding it to Cargo.toml is that devs will get a helpful error message if they have not updated

**Pull request merged**